rust/blockstore/src/arrow/migrations.rs

```diff
@@ -0,0 +1,81 @@
+use std::collections::HashSet;
+
+use super::{
+    provider::BlockManager,
+    root::{RootWriter, Version},
+    sparse_index::SetCountError,
+};
+use chroma_error::{ChromaError, ErrorCodes};
+use thiserror::Error;
+use uuid::Uuid;
+
+#[derive(Error, Debug)]
+pub enum MigrationError {
+    #[error("Block not found")]
+    BlockNotFound,
+    #[error("Block could not be fetched")]
+    BlockFetchError,
+    #[error("Error setting count")]
+    SetCountError(#[from] SetCountError),
+}
+
+impl ChromaError for MigrationError {
+    fn code(&self) -> ErrorCodes {
+        match self {
+            MigrationError::BlockNotFound => ErrorCodes::Internal,
+            MigrationError::BlockFetchError => ErrorCodes::Internal,
+            MigrationError::SetCountError(e) => e.code(),
+        }
+    }
+}
+
+async fn migrate_v1_to_v1_1(
+    root: &mut RootWriter,
+    block_manager: &BlockManager,
+    new_block_ids: &HashSet<Uuid>,
+) -> Result<(), MigrationError> {
+    if root.version == Version::V1 {
```

Comment:
Can we preserve the data migration comment, especially under refactor 

// MIGRATION(10/15/2024 @hammadb) Get all the blocks and manually update the sparse index